tools/clang/lib/SPIRV/SpirvEmitter.cpp

```diff
@@ -671,8 +671,9 @@ void SpirvEmitter::HandleTranslationUnit(ASTContext &context) {
   if (!spirvOptions.disableValidation) {
     std::string messages;
     if (!spirvToolsValidate(targetEnv, spirvOptions,
-                            declIdMapper.requiresLegalization(), &m,
-                            &messages)) {
+                            beforeHlslLegalization ||
+                                declIdMapper.requiresLegalization(),
```

Comment:
maybe you need to add a `beforeHlslLegalization` parameter to `spirvToolsValidate`